core/services/log/broadcaster.go

```diff
@@ -306,8 +319,14 @@ func (b *broadcaster) eventLoop(chRawLogs <-chan types.Log, chErr <-chan error)
 			needsResubscribe = b.onRmSubscribers() || needsResubscribe
 
 		case <-debounceResubscribe.C:
-			if needsResubscribe {
-				logger.Debug("LogBroadcaster: returning from the event loop to resubscribe")
+			restartFrom := atomic.LoadInt64(&b.restartFromNumber)
```

Comment:
maybe even cleaner than piggybacking on the debounceResubscribe and using an atomic int - is we could just use a restart channel and a new case here?

Reply:
I think that way we could get rid of restartFromNumber and just use requestedBackfillFrom

Reply:
ah yeah it's definitely a more idiomatic go - thanks!